### PR TITLE
SiteAddressChanger: Feature-wide copy updates

### LINKED ISSUE #22906: SiteRenamer: Full feature copy review.
We recently merged our feature branch for the SiteRenamer component & feature.
As this branch was growing quite large and parts of it were subject to change, we decided to avoid wrapping copy in `translate` and merged the feature without worrying about finalising copy right away.

Now that it's merged and it's functionality is likely not going to change dramatically any time soon we're now gearing up for flagging this as a production feature. In order to do that we're looking to finalise our copy feature-wide and would love the input of our word-smiths!

I've ran through the most likely flows of the feature and provided screenshots below.
I'll also link to the rest of the harder-to-reach copy in the code and give some context for each of those phrases.
TIA!

#### Site Address Rename Form in situ @ http://calypso.localhost:3000/domains/manage/site-name/edit/site-name
screen shot 2018-03-01 at 14 04 17

Copy Items:
- 'Edit Site Address' - Form/feature title
- 'Once changed, the current site address %(currentDomainName)s will no longer be available.' - info/warning
- 'Change Site Address' - Form Submit button (not consistent with form title)

#### Input Error, site address contains invalid characters
screen shot 2018-03-01 at 14 04 40

- 'Site address can only contain letters (a-z) and numbers.' - Validation error

#### Input Error, site address length is too short/too long
screen shot 2018-03-01 at 14 04 50

- 'Site address length should be between %(minimumLength)s and %(maximumLength)s characters.' - Validation error

#### Confirmation Dialog
screen shot 2018-03-01 at 14 05 12

- 'Cancel' - Dialog Action Button
- 'Change Site Address' - Dialog Action Button (again, not consistent with form title)
- "Let's Review" - Dialog title
- 'You are about to change your site address. Once changed, your previous site address will be unavailable for you or anyone else.' - Warning
- ... 'Will be removed and unavailable for use.' - Detail, follows the old site address. (Could use context property for `translate()` call.
- ... 'Will be your new site address.' - Detail, follows the new site address. (Could use context property for `translate()` call.
- 'Check the box to confirm' - Action instruction
- "I've double-checked and understand there is no undo." - Confirmation checkbox label

#### Successfully rename a site address:
screen shot 2018-03-01 at 14 06 16

'Your new domain name is ready to go!' - Success Notice (domain is not consistent with other language)

#### Site Address is already taken
screen shot 2018-03-01 at 14 05 25

'Sorry, that site has already been used.' - Error notice, perhaps 'site address' is better than just 'site'

There are more errors that originate from the server, I'll update this list once I have them all.

### PR DESCRIPTION
**Pull request by spen**
This PR puts some of the copy suggestions made in #22906 (those relevant to the front-end) in place in the Site Address Changer feature.

There's no logical change here so please just read and verify the copy updates included here :)

I've tagged this as needing a copy review - I'm sure we'd be happy for a second opinion, but @benhuberman is already on the case regarding copy :)

Fixes #22906

### CODE REVIEW

File: client/blocks/simple-site-rename-form/dialog.jsx [hunk 1]
```diff
@@ -89,11 +89,11 @@ class SiteRenamerConfirmationDialog extends PureComponent {
...
 					eventProperties={ { new_domain: newDomainName } }
 				/>
-				<h1>{ translate( "Let's Review" ) }</h1>
+				<h1>{ translate( "Let's review…" ) }</h1>
 				<p>
 					{ translate(
-						'You are about to change your site address. Once changed, ' +
-							'your previous site address will be unavailable for you or anyone else.'
+						"You're about to change your site address. Once you confirm the change, " +
+							'this site address will no longer be available for future use.'
```
Comment (frontdevde) [inline client/blocks/simple-site-rename-form/dialog.jsx]: Minor: We're mixing `"` and `'` here.

Comment (lsinger) [reply client/blocks/simple-site-rename-form/dialog.jsx]: IIRC this is the only way to do this without the linter complaining and thus failing the build.

Comment (frontdevde) [reply client/blocks/simple-site-rename-form/dialog.jsx]: TIL 👍

File: client/blocks/simple-site-rename-form/dialog.jsx [hunk 2]
```diff
@@ -130,7 +130,9 @@ class SiteRenamerConfirmationDialog extends PureComponent {
 						checked={ this.state.isConfirmationChecked }
 						onChange={ this.toggleConfirmationChecked }
 					/>
-					<span>{ translate( "I've double-checked and understand there is no undo." ) }</span>
+					<span>
+						{ translate( "I understand that I can't undo this change to my site address." ) }
```
Comment (andfinally) [inline client/blocks/simple-site-rename-form/dialog.jsx]: I feel like `I understand that I won't be able to undo this change to my site address.` might be more accurate, because at this point it hasn't happened yet.

File: client/blocks/simple-site-rename-form/dialog.jsx (hunk 1 above)
Comment (spen) [reply client/blocks/simple-site-rename-form/dialog.jsx]: Yah, just because of the `You're` 🤷‍♂️

File: client/blocks/simple-site-rename-form/dialog.jsx (hunk 2 above)
Comment (spen) [reply client/blocks/simple-site-rename-form/dialog.jsx]: Hmm, I see what you're saying! There's context since we're in the dialog already but your suggestion may still be a touch more explicit - @benhuberman, do you have any thoughts on this?

File: client/blocks/simple-site-rename-form/dialog.jsx (hunk 1 above)
Comment (frontdevde) [reply client/blocks/simple-site-rename-form/dialog.jsx]: Oh right, didn't see the `You're` 🙈
👮‍♀️*move along, nothing to see here*

File: client/blocks/simple-site-rename-form/dialog.jsx (hunk 2 above)
Comment (spen) [reply client/blocks/simple-site-rename-form/dialog.jsx]: Sweet! Update made - I'll merge once checks are ready :)
